src/SensorFactory.cc

```diff
@@ -75,21 +72,31 @@ std::shared_ptr<SensorPlugin> SensorFactory::LoadSensorPlugin(
     return std::shared_ptr<SensorPlugin>();
   }
 
-  auto pluginNames = this->dataPtr->pluginLoader.LoadLibrary(fullPath);
+  plugin::Loader pluginLoader;
+  auto pluginNames = pluginLoader.LoadLib(fullPath);
```

Comment:
what do you think of using the `Loader::PluginsImplementing` class to preselect the plugins that give the desired interface?

~~~
diff --git a/src/SensorFactory.cc b/src/SensorFactory.cc
index 99e2de4..48956d3 100644
--- a/src/SensorFactory.cc
+++ b/src/SensorFactory.cc
@@ -73,7 +73,9 @@ std::shared_ptr<SensorPlugin> SensorFactory::LoadSensorPlugin(
   }
 
   plugin::Loader pluginLoader;
-  auto pluginNames = pluginLoader.LoadLib(fullPath);
+  pluginLoader.LoadLib(fullPath);
+  auto pluginNames =
+      pluginLoader.PluginsImplementing<ignition::sensors::SensorPlugin>();
 
   if (pluginNames.empty())
   {
~~~